### PR TITLE
Use babel instead of deprecated react-tools

### PR DESCRIPTION
**Pull request by benmccann**
Fixes https://github.com/lagom/activator-lagom-java-chirper/issues/56

`react-tools` is deprecated. It's sure to be only a matter of time until using it causes problems. Using babel is a much more standard way of setting up the project and so is much more useful to people who want to see what a project setup actually looks like in real life

Takes an extra 6s to setup the very first time you run the app. 4s faster every subsequent time you do a clean compile because it stops storing the node modules in the target directory where they're easily wiped out and instead uses the standard location

### CODE REVIEW

**Comment by TimMoore** [comment]
@fsat what do you think?

**Comment by TimMoore** [comment]
@benmccann are you using Windows (or have you tested on Windows)?

**Comment by benmccann** [comment]
No, unfortunately I don't have a Windows machine to test on. However, I made sure to use `node_modules` just as @fsat is doing in his PR, so hopefully this should work on Windows

**Comment by TimMoore** [comment]
@megadotnet are you able to give this branch a try?

**Comment by fsat** [comment]
@TimMoore - I think this is great! If `babeljs` does work on Windows and this PR is merged, then #55 should be closed.

Thanks @benmccann - hopefully `babeljs` works on Windows, and if it does, then this PR should resolve #54 as well.

**Comment by TimMoore** [comment]
I'll try to test this in a Windows VM tomorrow, if @megadotnet doesn't get back to us first.

**Comment by TimMoore** [comment]
I was able to verify the bug reported in #54 in a [Windows 10 VM](https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/)

Unfortunately, this branch introduces a new problem:

virtualbox_msedge - win10_preview_18_05_2017_19_14_20

**Comment by TimMoore** [comment]
At the risk of shaving a giant JavaScript yak, maybe the best bet would be to set up a Grunt, Gulp or WebPack build and use the `maven-frontend-plugin` ability to invoke one of those.

**Comment by benmccann** [comment]
Setting up WebPack could be nice. I've never used it before, but it seems like it's becoming the defacto, so it would be cool to show an example. Unfortunately, it would require a bit more re-work. Right now we produce a `.js` file for each `.jsx`. WebPack would bundle everything as a single file. And that's the better way to do things, but is an additional change. And WebPack would still execute Babel, so it probably doesn't hurt to get Babel working first.

**Comment by benmccann** [comment]
@TimMoore would you be able to test on Windows again? That error you got should be fixed now

**Comment by TimMoore** [comment]
Still not quite there:

```
java.io.IOException: Cannot run program "node" (in directory "C:\Users\IEUser\activator-lagom-java-chirper\front-end"): CreateProcess error=2, The system cannot find the file specified
```

File: front-end/pom.xml [hunk 1]
```diff
@@ -184,14 +183,14 @@
                         <goal>exec</goal>
                     </goals>
                     <configuration>
-                        <executable>${basedir}/target/node/bin/jsx</executable>
+                        <executable>node</executable>
                         <environmentVariables>
                             <PATH>${env.PATH}:${pom.basedir}/target/node/node</PATH>
```
Comment (TimMoore) [inline front-end/pom.xml]: This `PATH` setting doesn't help to find the executable itself, it's only passed to the child process AFAICT.

**Comment by TimMoore** [comment]
The reason I was suggesting WebPack is that the `frontend-maven-plugin` already handles the platform differences internally. Unfortunately, they don't expose a goal for executing arbitrary commands installed by `npm`, which would be really useful here.

Fortunately, you can pass arbitrary arguments to `npm`, and `npm` supports a `run-script` argument that lets you run a script defined in `package.json`. I'll see if I can get that to work.

File: README.md [hunk 2]
```diff
@@ -3,3 +3,4 @@
 
 A Lagom Java template for Lightbend Activator showcasing a twitter-like application
 
+Start all services using `mvn lagom:runAll`.
```
Comment (TimMoore) [inline README.md]: sbt is also still supported, so how about this?

> Start all services using `mvn lagom:runAll` or `sbt runAll`

**Comment by TimMoore** [comment]
@benmccann I took the liberty of pushing a change to your branch. Hope that's OK. This runs babel via `npm run-script` and I have tested that it works when I run `mvn lagom:runAll` (including hot reload when I change a `jsx` file) on Windows 10 and on my Mac.

I also checked `sbt runAll` and it seems to be all good as well.

If you're happy with my changes, I'll go ahead and merge.

Thanks again for getting the ball rolling on this.

**Comment by benmccann** [comment]
Awesome! Thank you @TimMoore !

Yes, please go ahead and merge :-)

**Pull request merged**